go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -841,6 +838,48 @@ func (tm *TabletManager) handleRelayLogError(err error) error {
 	return err
 }
 
+// repairReplication tries to connect this server to whoever is
+// the current master of the shard, and start replicating.
+func (tm *TabletManager) repairReplication(ctx context.Context) error {
+	tablet := tm.Tablet()
+
+	si, err := tm.TopoServer.GetShard(ctx, tablet.Keyspace, tablet.Shard)
+	if err != nil {
+		return err
+	}
+	if !si.HasMaster() {
+		return fmt.Errorf("no master tablet for shard %v/%v", tablet.Keyspace, tablet.Shard)
```

Comment:
Any reason this file is using fmt.Error instead of vterrors?

Reply:
I only moved this code from somewhere else. All these older functions are still using fmt.Errorf.

Reply:
I could not see any test file for this.

Reply:
There are end2end tests for this. We should ideally have unit tests also, but it's legacy code.